HEN_HOUSE/egs++/geometry/egs_prism/egs_prism.h

```diff
@@ -221,7 +221,11 @@ class EGS_PRISM_EXPORT EGS_PrismT : public EGS_BaseGeometry {
                 tt = 0;
             }
             if (tt <= t) {
-                t = tt;
+                if (tt > 0) {
+                    t = tt+boundaryTolerance;
```

Comment:
Should this be `halfBoundaryTolerance` as in the nd geometry fix above?

Reply:
So far I have only found the need to apply `halfBoundaryTolerance` in cases where you are setting `t = 0` or `t = boundaryTolerance`. That's when you can run into issues in the CD. Those cases were in `egs_cylinders` and `egs_spheres`. Maybe this is something you can tackle more generally from the CD side, but I found that much trickier.

I do also spot `t = 0` and `d = 0` in `egs_cones` and `egs_planes`, so it might be worth hunting for geometry errors with those geometries in a CD.